odoo/service/model.py

```diff
@@ -153,7 +105,13 @@ def retrying(func, env):
                         else:
                             raise RuntimeError(f"Cannot retry request on input file {filename!r} after serialization failure") from exc
                 if isinstance(exc, IntegrityError):
-                    raise _as_validation_error(env, exc) from exc
+                    model = env['base']
+                    for rclass in env.registry.values():
+                        if exc.diag.table_name == rclass._table:
+                            model = env[rclass._name]
+                            break
+                    message = env._("The operation cannot be completed: %s", model._sql_error_to_message(exc))
```

Comment:
Here `model.env.cr` is a broken transaction.  Therefore `model._sql_error_to_message(exc)` will not be able to make SQL queries.

Reply:
There is a `env.cr.rollback()` at the beginning of the except block.

Reply:
Oops, I missed that.